File: src/models/MenuItem.ts
import { Category } from "./Categories";
import { IMenuItem } from "./interfaces/IMenuItem";

export class MenuItem implements IMenuItem {
  name: string;
  imageUrl?: string | undefined;
  description?: string | undefined;
  price: number;
  categories?: Category[] | undefined;
  isFavorite?: boolean | undefined;
  isActive?: boolean | undefined;

  constructor(private menuItem: IMenuItem) {
    this.name = menuItem.name;
    this.imageUrl = menuItem.imageUrl;
    this.description = menuItem.description;
    this.price = menuItem.price;
    this.categories = menuItem.categories;
    this.isFavorite = menuItem.isFavorite;
    this.isActive = menuItem.isActive;
  }

  hasDescription() {
    return typeof this.description === "string" && this.description.length > 0;
  }
}

File: haru_cafe_django/haru_cms/models/MenuItem.py
from django.db import models

class MenuItem(models.Model):
  name = models.CharField(max_length=256)
  image = models.ImageField(upload_to="haru_cms/images", blank=True, null=True)
  description = models.CharField(max_length=1024, blank=True, null=True)
  price = models.DecimalField(max_digits=10, decimal_places=2)
  categories = models.ManyToManyField('MenuItemCategory')
  isFavorite = models.BooleanField(default=False)
  isActive = models.BooleanField(default=True)
  size = models.ForeignKey('MenuItemSize', on_delete=models.DO_NOTHING, blank=True, null=True)

  def __str__(self):
    return self.name

